linked_list/node.py

```diff
@@ -4,13 +4,19 @@ class Node(object):
     """
 
     def __init__(self, elem, next=None):
-        pass
+        self.elem = elem
+        self.next = next
 
     def __str__(self):
-        pass
+        return str(self.elem)
 
     def __eq__(self, other):
-        pass
+        
+        #if self.elem is not None and other.elem is not None:
+        if type(other) is Node:
+            return self.elem == other.elem
+        else:
+            return self.elem == other
```

Comment:
In our pop method we were returning the popped node rather than the value (elem) held by the popped node. In the tests however the node returned by pop is compared to an int (which is why we allow for Node(3) == 3). I think you're correct however that Node(3) == 3 should be false so instead I've fixed it and our pop method now returns the popped_node.elem instead of the popped_node. Thanks!